Dockerfile

```diff
@@ -1,44 +1,127 @@
+#
+# Basic Parameters
+#
+ARG ARCH="amd64"
+ARG OS="linux"
+ARG VER="2021.03.11"
+ARG PKG="cloudconfig"
+ARG SRC="https://github.com/ArkCase/acm-config-server.git"
+
+FROM 345280441424.dkr.ecr.ap-south-1.amazonaws.com/ark_base:latest as src
+
+#
+# Basic Parameters
+#
+ARG ARCH
+ARG OS
+ARG VER
+ARG PKG
+ARG SRC
+ARG MVN_VER="3.8.6"
+ARG MVN_SRC="https://dlcdn.apache.org/maven/maven-3/${MVN_VER}/binaries/apache-maven-${MVN_VER}-bin.tar.gz"
+
+LABEL ORG="ArkCase LLC"
+LABEL MAINTAINER="Armedia Devops Team <devops@armedia.com>"
+LABEL APP="Cloudconfig"
+LABEL VERSION="${VER}"
+
+# Environment variables
+ENV VER="${VER}"
+ENV SRC="${SRC}"
+ENV JAVA_HOME="/usr/lib/jvm/java"
+ENV LANG="en_US.UTF-8"
+ENV LANGUAGE="en_US:en"
+ENV LC_ALL="en_US.UTF-8"
+ENV MVN_VER="3.8.6"
+ENV MVN_SRC="https://dlcdn.apache.org/maven/maven-3/${MVN_VER}/binaries/apache-maven-${MVN_VER}-bin.tar.gz"
+
+WORKDIR "/src"
+
+# First, install the JDK
+RUN yum update -y && yum -y install java-1.8.0-openjdk-devel git && yum clean all
+
+# Next, the stuff that will be needed for the build
+COPY "mvn" "/usr/bin"
+ADD "${MVN_SRC}" "/"
+RUN echo "Installing Maven ${MVN_VER}..." && tar -C / -xzf "/apache-maven-${MVN_VER}-bin.tar.gz" && mv -vf "/apache-maven-${MVN_VER}" "/mvn"
+RUN echo "Cloning version [${VER}] from [${SRC}]..." && git clone -b "${VER}" "${SRC}" . && ls -l && mvn clean verify
+
+########################################
+# Build ConfigServer                   #
+########################################
+
 FROM 345280441424.dkr.ecr.ap-south-1.amazonaws.com/ark_base:latest
```

Comment:
The first container is used to do the build. Then, the built artifacts are copied onto the 2nd (final) container. This way we can have one container swallow up all the on-disk overhead required for building, and then just pluck what we're interested in onto the final container.

We do this as well in the Samba container since we need to rebuild it with a flag that enables AD emulation support.

This is fine ;)